packages/create-app/src/lib/bin.ts

```diff
@@ -65,8 +69,12 @@ export async function run() {
 
   printWelcomeMessage();
 
-  const projectName =
+  let projectName =
     (options.dir || options.name) ?? (await promptProjectName());
+  if (validateProjectName(projectName)) {
+    projectName = await promptProjectName(projectName);
```

Comment:
`validateProjectName` may return a string with error message. I think it would be weird if it ended up as a suggestion for a name in this prompt?